### ISSUE TITLE
internal/grpc: TestGRPCStreaming/StreamRoutes test flake

### ISSUE THREAD

**Issue by davecheney** [heptio/contour#327]
```
--- FAIL: TestGRPCStreaming (0.40s)
	grpc_test.go:338: time="2018-04-17T18:44:38Z" level=info msg="waiting for notification" api=RDS connection=1 version=0
	grpc_test.go:338: time="2018-04-17T18:44:38Z" level=info msg="notification received" api=RDS connection=1 version=1
	grpc_test.go:338: time="2018-04-17T18:44:38Z" level=info msg="waiting for notification" api=RDS connection=1 version=1
	grpc_test.go:338: time="2018-04-17T18:44:38Z" level=error msg="stream terminated" api=RDS connection=1 error="context deadline exceeded"
    --- FAIL: TestGRPCStreaming/StreamRoutes (0.10s)
    	grpc_test.go:186: expected "DeadlineExceeded", got "Unknown"
	grpc_test.go:338: time="2018-04-17T18:44:38Z" level=info msg="waiting for notification" api=CDS connection=1 version=0
	grpc_test.go:338: time="2018-04-17T18:44:38Z" level=info msg="notification received" api=CDS connection=1 version=1
	grpc_test.go:338: time="2018-04-17T18:44:38Z" level=info msg="waiting for notification" api=CDS connection=1 version=1
	grpc_test.go:338: time="2018-04-17T18:44:38Z" level=error msg="stream terminated" api=CDS connection=1 error="context deadline exceeded"
	grpc_test.go:338: time="2018-04-17T18:44:38Z" level=info msg="waiting for notification" api=EDS connection=1 version=0
	grpc_test.go:338: time="2018-04-17T18:44:38Z" level=info msg="notification received" api=EDS connection=1 version=1
	grpc_test.go:338: time="2018-04-17T18:44:38Z" level=info msg="waiting for notification" api=EDS connection=1 version=1
	grpc_test.go:338: time="2018-04-17T18:44:38Z" level=error msg="stream terminated" api=EDS connection=1 error="context canceled"
	grpc_test.go:338: time="2018-04-17T18:44:38Z" level=info msg="waiting for notification" api=LDS connection=1 version=0
	grpc_test.go:338: time="2018-04-17T18:44:38Z" level=info msg="notification received" api=LDS connection=1 version=1
	grpc_test.go:338: time="2018-04-17T18:44:38Z" level=info msg="waiting for notification" api=LDS connection=1 version=1
	grpc_test.go:338: time="2018-04-17T18:44:38Z" level=error msg="stream terminated" api=LDS connection=1 error="context deadline exceeded"
	grpc_test.go:338: time="2018-04-17T18:44:38Z" level=info msg="waiting for notification" api=LDS connection=2 version=0
	grpc_test.go:338: time="2018-04-17T18:44:38Z" level=info msg="notification received" api=LDS connection=2 version=1
	grpc_test.go:338: time="2018-04-17T18:44:38Z" level=info msg="waiting for notification" api=LDS connection=2 version=1
	grpc_test.go:338: time="2018-04-17T18:44:38Z" level=error msg="stream terminated" api=LDS connection=2 error="context canceled"
FAIL
FAIL	github.com/heptio/contour/internal/grpc	0.458s
```

**Comment by rosskukulinski** [comment]
@davecheney is this still an issue?

**Comment by davecheney** [comment]
Bumping to 0.7, related to #311

**Comment by davecheney** [comment]
Fixed in #756
